### ISSUE TITLE
Allow commands to have their own location

### ISSUE THREAD

**Issue by Void-E** [DannyBen/bashly#170]
When using bashly for larger cli projects, the src folder can be a mess. 

For now I'm using more than 50 different commands. Some with splitted configs, but all SH files are located in the `src` directory. 

It would be cool if there is an option to provide a custom path per command, so I can handle my own script structure.

After diggin' the source code and docs, I couldn't find any option for that.

**Comment by DannyBen** [comment]
and in your scenario, what about subcommands? I am assuming one would expect subcommands to also inherit the `source` attribute form their parent, unless explicitly overridden, right? (which complicates the potential solution a bit more).

**Comment by Void-E** [comment]
Yep, that's exactly what I was thinking about. And I agree that it would complicate it (maybe too) much more.

**Comment by DannyBen** [comment]
Well - its a good suggestion nonetheless. I am also in favor of files neatly organized and tucked in their own folder.

Let me think about it a little bit. Perhaps something like this can be done by modifying this area of code:

https://github.com/DannyBen/bashly/blob/f0a59f4e2452034a68ad85f7d6c0d55ffe430d06/lib/bashly/script/command.rb#L55-L59

While I evaluate the options for implementations, feel free to add your own thoughts here.

**Comment by DannyBen** [comment]
I can add an argument `filename` which will replace the default filename if present:

```yaml
- name: upload
  short: u
  help: Upload a file
  filename: commands/upload.sh   # this is still under `src`
```

This will NOT inherit anything - each command without a filename will be generated and expected to be in its default place.

This change I can implement immediately. Is this an acceptable solution in your view (@Void-E)?

**Comment by DannyBen** [comment]
@Void-E - the `filename` change is ready to be merged. Can you confirm it covers your use case?

**Comment by Void-E** [comment]
Yeah, this will absolutely covers my use case!

**Comment by DannyBen** [comment]
This is now implemented in v0.7.3, thanks for the suggestion.

**Comment by DannyBen** [comment]
This was raised again in #398 - so I am adding a new setting (in `settings.yml`) to automatically organize command files in command subdirectories (#399).
